Sources/UAPusher/Builder/UABuilder.swift

```diff
@@ -0,0 +1,245 @@
+    ///
+    /// - Parameter predefined: Options.Predefined
+    /// - Returns: UABuilder
+    /// - Throws: In case of invalid JSON
+    public func options(predefined: Options.Predefined) throws -> UABuilder {
+        self.optionsPayload = try Options(predefined: predefined)
+        return self
+    }
+    
+    // MARK: InApp
+    
+    /// Set in_app from a JSON object
+    ///
+    /// - Parameter inApp: JSON
+    /// - Returns: UABuilder
+    public func inApp(inApp: JSON) -> UABuilder {
+        self.inAppPayload = InApp(payload: inApp)
+        return self
+    }
+    
+    /// Set in_app from a predefined value
+    ///
+    /// - Parameter predefined: InApp.Predefined
+    /// - Returns: UABuilder
+    /// - Throws: In case of invalid JSON
+    public func inApp(predefined: InApp.Predefined) throws -> UABuilder {
+        self.inAppPayload = try InApp(predefined: predefined)
+        return self
+    }
+    
+    // MARK: Message
+    
+    /// Set message from a JSON object
+    ///
+    /// - Parameter message: JSON
+    /// - Returns: UABuilder
+    public func message(message: JSON) -> UABuilder {
+        self.messagePayload = Message(payload: message)
+        return self
+    }
+    
+    /// Set message from a predefined value
+    ///
+    /// - Parameter predefined: InApp.Predefined
+    /// - Returns: UABuilder
+    /// - Throws: In case of invalid JSON
+    public func message(predefined: Message.Predefined) throws -> UABuilder {
+        self.messagePayload = try Message(predefined: predefined)
+        return self
+    }
+    
+    // MARK: Campaigns
+    
+    /// Set campaigns from a JSON object
+    ///
+    /// - Parameter campaigns: JSON
+    /// - Returns: UABuilder
+    public func campaigns(campaigns: JSON) -> UABuilder {
+        self.campaignsPayload = Campaigns(payload: campaigns)
+        return self
+    }
+    
+    /// Set campaigns from a predefined value
+    ///
+    /// - Parameter predefined: Campaign.Predefined
+    /// - Returns: UABuilder
+    /// - Throws: In case of invalid json
+    public func campaigns(predefined: Campaigns.Predefined) throws -> UABuilder {
+        self.campaignsPayload = try Campaigns(predefined: predefined)
+        return self
+    }
+    
+    /// Set campaigns from a list of categories
+    ///
+    /// - Parameter categories: [String]
+    /// - Returns: UABuilder
+    /// - Throws: In case campaig json cannot be created
+    public func campaigns(categories: [String]) throws -> UABuilder {
+        self.campaignsPayload = try Campaigns(categories: categories)
+        return self
+    }
+    
+    // MARK: Build
+    
+    /// Build the payload
+    ///
+    /// - Returns: JSON
+    /// - Throws: In case of invalid JSON
+    public func makeBody() throws -> JSON {
+        var body: JSON = JSON()
+        
+        if let unwrappedAudience: Audience = audiencePayload {
+            try body.set("audience", unwrappedAudience.payload)
```

Comment:
Should we move these keys to the building blocks to keep that logic more grouped/isolated?